File: Remove Duplicates from Sorted List/Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        ListNode* ptr = head;
        ListNode* fastPtr = NULL;
        // Empty list
        if(!ptr)    return head;
        // fastPtr always points to ptr->next
        if(ptr->next)   fastPtr = ptr->next;
        // Compare ptr and ptr->next
        // If they've save values, link ptr to ptr->next->next
        while(fastPtr){
            if(ptr->val == fastPtr->val){
                // Not reaching the end yet
                if(fastPtr->next){
                    ptr->next = fastPtr->next;
                    fastPtr = fastPtr->next;
                }
                // Reaches the end
                else{
                    ptr->next = NULL;
                    fastPtr = NULL;
                    return head;
                }
            }
            else{
                ptr = ptr->next;
                fastPtr = fastPtr->next;
            }
        }
        return head;
    }
};

File: Remove Duplicates from Sorted List/Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null)            return head;
        if(head.next == null)       return head;
        ListNode fastNode = head.next;
        ListNode Node = head;
        
        while(fastNode != null){
            if(fastNode.val == Node.val){
                fastNode = fastNode.next;
                Node.next = fastNode;
            }
            else{
                fastNode = fastNode.next;
                Node = Node.next;
            }
        }
        
        return head;
    }
}
